MdeModulePkg/Core/Dxe/Misc/MemoryProtectionSupport.c

```diff
@@ -502,15 +672,30 @@ CollectSpecialRegionHobs (
   VOID
   )
 {
-  EFI_HOB_GUID_TYPE                            *GuidHob          = NULL;
-  MEMORY_PROTECTION_SPECIAL_REGION             *HobSpecialRegion = NULL;
-  MEMORY_PROTECTION_SPECIAL_REGION_LIST_ENTRY  *NewSpecialRegion = NULL;
+  EFI_HOB_GUID_TYPE                            *GuidHob                  = NULL;
+  MEMORY_PROTECTION_SPECIAL_REGION             *HobSpecialRegion         = NULL;
+  MEMORY_PROTECTION_SPECIAL_REGION_LIST_ENTRY  *NewSpecialRegion         = NULL;
+  MEMORY_PROTECTION_SPECIAL_REGION_LIST_ENTRY  *OverlappingSpecialRegion = NULL;
 
   GuidHob = GetFirstGuidHob (&gMemoryProtectionSpecialRegionHobGuid);
 
   while (GuidHob != NULL) {
-    HobSpecialRegion = (MEMORY_PROTECTION_SPECIAL_REGION *)GET_GUID_HOB_DATA (GuidHob);
-    NewSpecialRegion = AllocateCopyPool (sizeof (MEMORY_PROTECTION_SPECIAL_REGION_LIST_ENTRY), HobSpecialRegion);
+    HobSpecialRegion         = (MEMORY_PROTECTION_SPECIAL_REGION *)GET_GUID_HOB_DATA (GuidHob);
+    OverlappingSpecialRegion = NULL;
+    if (DoesSpecialRegionConflict (HobSpecialRegion, &mSpecialMemoryRegionsPrivate.SpecialRegionList)) {
+      DEBUG ((
+        DEBUG_ERROR,
+        "%a - Special region 0x%llx - 0x%llx conflicts with another special region!\n",
+        __FUNCTION__,
+        HobSpecialRegion->Start,
+        HobSpecialRegion->Start + HobSpecialRegion->Length
+        ));
+      ASSERT (FALSE);
+      GuidHob = GetNextGuidHob (&gMemoryProtectionSpecialRegionHobGuid, GET_NEXT_HOB (GuidHob));
+      continue;
+    }
+

```

Comment:
nit style: I think the continue can be removed and everything can just be an else statement?